pkg/packing/packable.go

```diff
@@ -100,10 +101,29 @@ func (p *Packable) Pack(pods []*v1.Pod) *Result {
 			return result
 		}
 		result.unpacked = append(result.unpacked, pod)
+
+		if p.isFull(pods[len(pods)-1]) {
```

Comment:
Is it possible to move this check to the beginning of the loop? Unless I'm misunderstanding.

Reply:
it can be moved to the top. I think it might make more sense to move after the reserve block though. 

Here's how I reason in english about the func:

Try to reserve the space for the pod, if that fails, then check if the node is full and potentially exit the loop. If the node isn't full (next block) and nothing has been packed (meaning the largest pod was unable to be packed), exit the loop because we need to find a bigger node. If we've made progress, but the current node is just getting full, add the items to the unpacked list and keep trying smaller pods.